### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

### CODE REVIEW

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 1]
```diff
@@ -639,8 +639,8 @@ cell astronomy_celestial_object
     optional List<CellId> astronomy_celestial_object_category;
     optional float astronomy_celestial_object_cosmological_distance;
     optional float astronomy_celestial_object_magnitude;
-    optional float astronomy_celestial_object_decllnation;
-    optional float astronomy_celestial_object_right_ascention;
+    optional float astronomy_celestial_object_declination;
```
Comment (jsoref) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: I have no idea what this stuff is. It's possible it's third party content that shouldn't be touched. I can easily drop it if that's the case. I'm pretty sure my corrections are correct (assuming there aren't any foreign references that care about the spelling).

I also didn't fix all of the errors, some required more thought that I had the energy to invest.

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs [hunk 2]
```diff
@@ -272,9 +272,9 @@ public void JsonDSLTest2_11()
   },
   'paper': {
     'type': 'Paper',
-    'select': [ 'OriginalTitle', 'ConferneceID' ],
+    'select': [ 'OriginalTitle', 'ConferenceID' ],
```
Comment (jsoref) [inline src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: I hope this is just a test that doesn't care about the spelling...

File: src/Modules/LIKQ/FanoutSearch/Core/FanoutSearchModule.cs [hunk 3]
```diff
@@ -148,7 +148,7 @@ public static void RegisterExpressionSerializerFactory(Func<IExpressionSerialize
             ExpressionSerializer.SetSerializerFactory(func);
         }
 
-        public static void RegisterQueryWhilelistType(Type t)
+        public static void RegisterQueryWhitelistType(Type t)
```
Comment (jsoref) [inline src/Modules/LIKQ/FanoutSearch/Core/FanoutSearchModule.cs:151]: This could be an api change.

File: src/Modules/LIKQ/FanoutSearch/QueryLanguage/TraverseActionRewritter.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ namespace FanoutSearch
     ///       not accept a value that is not defined in the enumeration type, even if
     ///       the value is a combination of the flags defined in the type.
     /// </summary>
-    class TraverseActionRewritter : ExpressionVisitor
+    class TraverseActionRewriter : ExpressionVisitor
```
Comment (jsoref) [inline src/Modules/LIKQ/FanoutSearch/QueryLanguage/TraverseActionRewritter.cs:23]: This might be an API change.

File: src/Trinity.C/src/Storage/MTHash/MTHash.lock.cpp [hunk 5]
```diff
@@ -194,9 +194,9 @@ namespace Storage
 
         auto                   batch_get_lock_action = [&](std::atomic<uint64_t>* ptr)
         {
-            uint64_t comprand = BUCKETLOCK_EIGHT_NOLOCKS;
+            uint64_t comparand = BUCKETLOCK_EIGHT_NOLOCKS;
```
Comment (jsoref) [inline src/Trinity.C/src/Storage/MTHash/MTHash.lock.cpp:197]: I'm not sure about this. My change is just to make it match another instance elsewhere.

File: samples/freebase-likq/freebase-tsl/freebase.tsl (hunk 1 above)
Comment (yatli) [reply samples/freebase-likq/freebase-tsl/freebase.tsl]: these are automatically parsed from the `Freebase` data set. Please drop these changes (as they will be populated again when we refresh the dataset).

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs (hunk 2 above)
Comment (yatli) [reply src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: it's got the wrong spell from the beginning so we adapted to that in the test. :/

File: cmake/PostBuild.cmake [hunk 6]
```diff
@@ -99,6 +99,6 @@ FUNCTION(SET_OUTPUT_DIRECTORY dir)
     SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY_RELWITHDEBINFO ${dir} PARENT_SCOPE)
     SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_RELWITHDEBINFO ${dir} PARENT_SCOPE)
 
-    SET(OTUPUT_DIRECTORY_IS_SET TRUE PARENT_SCOPE)
+    SET(OUTPUT_DIRECTORY_IS_SET TRUE PARENT_SCOPE)
```
Comment (yatli) [inline cmake/PostBuild.cmake:102]: this one definitely fixed some weird build glitches... lol

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 7]
```diff
@@ -639,8 +639,8 @@ cell astronomy_celestial_object
...
+    optional float astronomy_celestial_object_right_ascension;
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert.

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 8]
```diff
@@ -788,7 +788,7 @@ cell astronomy_meteor_shower
 {
     optional u8string astronomy_meteor_shower_source_of_meteor_shower;
     [GraphEdge]
-    optional CellId astronomy_meteor_shower_common_calendar_occurance;
+    optional CellId astronomy_meteor_shower_common_calendar_occurrence;
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert.

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 9]
```diff
@@ -6315,7 +6315,7 @@ cell exhibitions_exhibition
 cell exhibitions_exhibition_curator
 {
     [GraphEdge]
-    optional List<CellId> exhibitions_exhibition_curator_exhibitions_curated;
+    optional List<CellId> exhibitions_exhibition_curator_exhibitions_currated;
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 10]
```diff
@@ -9092,10 +9092,10 @@ cell location_nl_province
     [GraphEdge]
     optional CellId location_nl_province_capital;
 }
-cell location_offical_symbol_variety
+cell location_official_symbol_variety
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 11]
```diff
@@ -9092,10 +9092,10 @@ cell location_nl_province
...
 {
     [GraphEdge]
-    optional List<CellId> location_offical_symbol_variety_symbols_of_this_kind;
+    optional List<CellId> location_official_symbol_variety_symbols_of_this_kind;
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert

File: samples/freebase-likq/freebase-tsl/freebase.tsl [hunk 12]
```diff
@@ -10621,7 +10621,7 @@ cell medicine_medical_trial
     [GraphEdge]
     optional List<CellId> medicine_medical_trial_diseases;
     [GraphEdge]
-    optional List<CellId> medicine_medical_trial_eligibile_genders;
+    optional List<CellId> medicine_medical_trial_eligible_genders;
```
Comment (yatli) [inline samples/freebase-likq/freebase-tsl/freebase.tsl]: please revert

File: src/Modules/GraphEngine.ServiceFabric/Trinity.ServiceFabric.GraphEngineRemotingClient/PackageRoot/ServiceManifest.xml [hunk 13]
```diff
@@ -19,7 +19,7 @@
     </EntryPoint>
   </CodePackage>
 
-  <!-- Config package is the contents of the Config directoy under PackageRoot that contains an 
+  <!-- Config package is the contents of the Config directory under PackageRoot that contains an
```
Comment (yatli) [inline src/Modules/GraphEngine.ServiceFabric/Trinity.ServiceFabric.GraphEngineRemotingClient/PackageRoot/ServiceManifest.xml:22]: I guess this is an upstream spell error from `Microsoft Service Fabric SDK`? ;)

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs (hunk 2 above)
Comment (yatli) [reply src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: so please revert this one. :)

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs [hunk 14]
```diff
@@ -272,9 +272,9 @@ public void JsonDSLTest2_11()
...
     'return': {
-      'has': 'ConferneceID'
+      'has': 'ConferenceID'
```
Comment (yatli) [inline src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: please revert.

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs [hunk 15]
```diff
@@ -358,8 +358,8 @@ public void JsonDSLTest2_15()
   },
   'paper': {
     'type': 'Paper',
-    'select': [ 'OriginalTitle', 'ConferneceID' ],
-    'return': { 'has': 'ConferneceID' }
+    'select': [ 'OriginalTitle', 'ConferenceID' ],
```
Comment (yatli) [inline src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: please revert.

File: src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs [hunk 16]
```diff
@@ -358,8 +358,8 @@ public void JsonDSLTest2_15()
...
+    'return': { 'has': 'ConferenceID' }
```
Comment (yatli) [inline src/Modules/LIKQ/FanoutSearch.UnitTest/JsonDSLTest2.cs]: please revert.

File: src/Modules/LIKQ/FanoutSearch/Core/FanoutSearchModule.cs (hunk 3 above)
Comment (yatli) [reply src/Modules/LIKQ/FanoutSearch/Core/FanoutSearchModule.cs:151]: no prob.

File: src/Modules/LIKQ/FanoutSearch/QueryLanguage/TraverseActionRewritter.cs (hunk 4 above)
Comment (yatli) [reply src/Modules/LIKQ/FanoutSearch/QueryLanguage/TraverseActionRewritter.cs:23]: looking good.

File: src/Trinity.C/src/Network/ProtocolConstants.h [hunk 17]
```diff
@@ -22,7 +22,7 @@ namespace Trinity
             PRESERVED_ASYNC,
         };
 
-        enum UInt32_Contants :uint32_t
+        enum UInt32_Constants :uint32_t
```
Comment (yatli) [inline src/Trinity.C/src/Network/ProtocolConstants.h]: could you please revert the changes in `Trinity.C/src/Network`? This part is heavily updated, here: https://github.com/Microsoft/GraphEngine/pull/227

File: src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.ContextObject.cpp [hunk 18]
```diff
@@ -62,15 +62,15 @@ namespace Trinity
             p->ReceivedPrefixBytes = 0;
             p->ReceivedMessageBodyBytes = 0;
 
-            p->RecvBuffer = (char*)malloc(UInt32_Contants::RecvBufferSize); // RecvBuffer should be read-only after being initialized
-            p->RecvBufferLen = UInt32_Contants::RecvBufferSize; // RecvBufferLen should be read-only after being initialized
-            p->avg_RecvBufferLen = UInt32_Contants::RecvBufferSize;
+            p->RecvBuffer = (char*)malloc(UInt32_Constants::RecvBufferSize); // RecvBuffer should be read-only after being initialized
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.ContextObject.cpp]: please revert.

File: src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Recv.cpp [hunk 19]
```diff
@@ -47,12 +47,12 @@ namespace Trinity
             }
 
             // Message prefix is not yet completely received
-            if (pContext->ReceivedPrefixBytes < UInt32_Contants::MessagePrefixLength)
+            if (pContext->ReceivedPrefixBytes < UInt32_Constants::MessagePrefixLength)
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Recv.cpp]: please revert.

File: src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Threading.cpp [hunk 20]
```diff
@@ -27,15 +27,15 @@ namespace Trinity
             Console::WriteLine("ServerSocket: Response: {0} {1}", messageBody, bytesToSend);
         }
 
-        // P/Inovke
+        // P/Invoke
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Threading.cpp]: please revert.

File: src/Trinity.C/src/Network/Server/posix/TrinitySocketServer.cpp [hunk 21]
```diff
@@ -62,9 +62,9 @@ namespace Trinity
         {
             PerSocketContextObject* p = (PerSocketContextObject*)malloc(sizeof(PerSocketContextObject));
 
-            p->RecvBuffer = (char*)malloc(UInt32_Contants::RecvBufferSize);
-            p->RecvBufferLen = UInt32_Contants::RecvBufferSize;
-            p->avg_RecvBufferLen = UInt32_Contants::RecvBufferSize;
+            p->RecvBuffer = (char*)malloc(UInt32_Constants::RecvBufferSize);
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/posix/TrinitySocketServer.cpp]: please revert.

File: src/Trinity.C/src/Storage/MTHash/MTHash.lock.cpp (hunk 5 above)
Comment (yatli) [reply src/Trinity.C/src/Storage/MTHash/MTHash.lock.cpp:197]: looking good.

File: src/Trinity.TSL/Trinity.TSL.CodeGen/CMakeLists.txt [hunk 22]
```diff
@@ -50,7 +50,7 @@ SET(metagen_out
     ${CMAKE_CURRENT_BINARY_DIR}/Libraries/StringAccessor.cs.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/Libraries/U8StringAccessor.cs.cpp
 
-    ${CMAKE_CURRENT_BINARY_DIR}/LINQ/ExpressionTreeRewritter.cs.cpp                                                                                                                                                                                                                     
+    ${CMAKE_CURRENT_BINARY_DIR}/LINQ/ExpressionTreeRewriter.cs.cpp
```
Comment (yatli) [inline src/Trinity.TSL/Trinity.TSL.CodeGen/CMakeLists.txt:53]: please revert this one unless the filename is also updated.

Comment (yatli) [reply src/Trinity.TSL/Trinity.TSL.CodeGen/CMakeLists.txt:53]: okay I see the file name changed down the page, so no need to revert :)

**Review by yatli — CHANGES_REQUESTED** [review]
Thanks for your contribution!

**Comment by jsoref** [comment]
@yatli: I've dropped `constants`, `conference`, `invoke`, and all changes to `freebase.tsl`.

**Comment by jsoref** [comment]
As for `Microsoft Service Fabric SDK`, can you report it to them?

**Comment by yatli** [comment]
@jsoref I will try to route this report.

**Comment by jsoref** [comment]
All the checks are happy, is there anything else you need from me?

File: src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Threading.cpp [hunk 23]
```diff
@@ -146,7 +146,7 @@ namespace Trinity
                 {
                     lstrcpynW(accept_sockaddr_strbuf, L"<Unknown address>", 1024);
                 }
-                Diagnostics::WriteLine(Diagnostics::LogLevel::Debug, "ServerSocket: Incomming connection from {0}", String(accept_sockaddr_strbuf));
+                Diagnostics::WriteLine(Diagnostics::LogLevel::Debug, "ServerSocket: Incoming connection from {0}", String(accept_sockaddr_strbuf));
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/iocp/TrinitySocketServer.Threading.cpp]: Please revert.

File: src/Trinity.C/src/Network/Server/posix/TrinitySocketServer.cpp [hunk 24]
```diff
@@ -118,7 +118,7 @@ namespace Trinity
                 {
                     strcpy(clientaddr, "Unknown address");
                 };
-                Diagnostics::WriteLine(Diagnostics::LogLevel::Debug, "ServerSocket: Incomming connection from {0}", String(clientaddr));
+                Diagnostics::WriteLine(Diagnostics::LogLevel::Debug, "ServerSocket: Incoming connection from {0}", String(clientaddr));
```
Comment (yatli) [inline src/Trinity.C/src/Network/Server/posix/TrinitySocketServer.cpp]: Please revert.

**Review by yatli — CHANGES_REQUESTED** [review]

**Comment by yatli** [comment]
I'm not seeing the new commit. Did you push?

**Comment by jsoref** [comment]
The tooling (hg) I use doesn't update the datestamps on commits I haven't changed.

GitHub will show your addressed commets as `outdated` and they will no longer be present in the PR.

What I'm doing in general is simply dropping commits (because each commit is a word and you're thankfully complaining about words).

As of this comment, I believe I've addressed all of your concerns.

**Review by yatli — APPROVED** [review]
I went over the file list again -- LGTM (the reverted files simply disappeared)

**Comment by yatli** [comment]
coming into `master` now. thanks again @jsoref.
btw that's an interesting idea -- to maintain spelling with a word-commit mapping.

**Comment by jsoref** [comment]
It makes dealing with reviews much easier. Even if someone only dislikes half of the changes for a given word, I can simply drop it for now and deal with it later. But as a general rule reviewers either like a word change or don't.

The other possibility is they don't like changes to certain files, and I deal with that by having a branch where I maintain commits that delete such files. Then I just rebase my fixes onto the branch to drop fixes for those files, and then rebase back onto master (excluding the ignore commits).

And of course, for long lasting reviews, it makes rebasing onto a moving master easier because any conflicts involve only a single family of misspellings, so fixing the commit is easier.
